crux-mir/src/Mir/Language.hs

```diff
@@ -122,17 +122,17 @@ defaultOutputConfig = Crux.defaultOutputConfig mirLoggingToSayWhat
 main :: IO ()
 main = do
     mkOutCfg <- defaultOutputConfig
-    exitCode <- mainWithOutputConfig mkOutCfg noExtraOverrides
+    exitCode <- mainWithOutputConfig (pure Nothing) mkOutCfg noExtraOverrides
     exitWith exitCode
 
-mainWithExtraOverrides :: BindExtraOverridesFn -> IO ()
-mainWithExtraOverrides bindExtra = do
+mainWithExtraOverrides :: (forall sym. IO (s sym)) -> BindExtraOverridesFn s -> IO ()
```

Comment:
This `forall sym. IO (s sym)` type appears in quite a few places to do similar things. I think the code would benefit if we packaged it up into a newtype, i.e., something like:

```hs
newtype InitUserState s = InitUserState (forall sym. IO (s sym))
```

Along with Haddocks that explain what it is used for. (For instance, when does the initialization occur?)